src/session.rs

```diff
@@ -2774,6 +2774,19 @@ impl Session {
                     Err(err) => self.message(format!("Error: {}", err), MessageType::Error),
                 }
             }
+            Command::SaveAs(ref path) => {
+                match self.active_view_mut().save_as(&Path::new(path).into()) {
+                    Ok(written) => {
+                        self.message(
+                            format!("\"{}\" {} pixels written", path, written),
+                            MessageType::Info,
+                        );
+                        self.active_view_mut().file_status =
+                            FileStatus::Saved(FileStorage::Single(Path::new(path).into()));
```

Comment:
This seemed the most straightforward way to change the view file name, but maybe there is some better way?